core/config/project_settings.cpp

```diff
@@ -340,16 +335,35 @@ bool ProjectSettings::_set(const StringName &p_name, const Variant &p_value) {
 bool ProjectSettings::_get(const StringName &p_name, Variant &r_ret) const {
 	_THREAD_SAFE_METHOD_
 
+	if (!props.has(p_name)) {
+		WARN_PRINT("Property not found: " + String(p_name));
+		return false;
+	}
+	r_ret = props[p_name].variant;
+	return true;
+}
+
+Variant ProjectSettings::get_setting_with_override(const StringName &p_name) const {
+	_THREAD_SAFE_METHOD_
+
 	StringName name = p_name;
 	if (!disable_feature_overrides && feature_overrides.has(name)) {
-		name = feature_overrides[name];
+		const LocalVector<Pair<StringName, StringName>> &overrides = feature_overrides[name];
+		for (uint32_t i = 0; i < overrides.size(); i++) {
+			if (OS::get_singleton()->has_feature(overrides[i].first)) { // Custom features are checked in OS.has_feature() already. No need to check twice.
+				if (props.has(overrides[i].second)) {
+					name = overrides[i].second;
+					break;
+				}
+			}
+		}
```

Comment:
It's not a new issue but it's worth noting that it's possible to have multiple feature overrides which are valid (e.g. `debug` and `pc`, or `mobile` and `android`), and it's not necessarily clear to the user / engine developers which one will be applied.

Just food for thought, I don't think we should change this behavior now, but it can lead to weird issues further down the road (especially if which one ends up first in the `features_overrides` vector is non deterministic.

Reply:
For now this is deterministic and still depends entirely of the order in which they are defined, but the inspector shows them alphabetically. This would need some changes in the Project Settings editor to detect overrides in the property list and sort them properly, as well as the ability to change their priority.

As this is not new, I would leave it like this for 4.1, but the code in ProjectSettings would need no change currently.